src/Statement/Filter/PropertySetStatementFilter.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Filter;
+
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementFilter;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class PropertySetStatementFilter implements StatementFilter {
+
+	/**
+	 * @var string[]
+	 */
+	private $propertyIds;
+
+	/**
+	 * @param string[]|string $propertyIds One or more property id serializations.
+	 */
+	public function __construct( $propertyIds ) {
```

Comment:
Might be a good use case for http://php.net/manual/en/class.splobjectstorage.php ?